openedx/core/djangoapps/waffle_utils/views.py

```diff
@@ -214,20 +217,53 @@ def _get_or_create_toggle_response(self, toggles_dict, toggle_name):
 
     def _get_settings_state(self):
         """
-        Returns a dictionary of settings values. Will only return values that are set to true or false.
+        Returns a dictionary of settings values. Will only return values that are set to true or false and for which a
+        corresponding SettingToggle or SettingDictToggle does not exist.
         """
+        setting_toggles = set(toggle.name for toggle in SettingToggle.get_instances())
+        setting_dict_toggles = set((toggle.name, toggle.key) for toggle in SettingDictToggle.get_instances())
 
         bool_settings = list()
         for setting_name, setting_value in vars(settings).items():
             if isinstance(setting_value, dict):
                 for dict_name, dict_value in setting_value.items():
-                    if isinstance(dict_value, bool):
+                    if isinstance(dict_value, bool) and (setting_name, dict_name) not in setting_dict_toggles:
                         bool_settings.append(
                             {
                                 'name': "{setting_name}['{dict_name}']".format(setting_name=setting_name, dict_name=dict_name),
                                 'is_active': dict_value,
                             }
                         )
             elif isinstance(setting_value, bool):
-                bool_settings.append({'name': setting_name, 'is_active': setting_value})
+                if setting_name not in setting_toggles:
+                    bool_settings.append({'name': setting_name, 'is_active': setting_value})
         return bool_settings
+
+    def _get_setting_toggle_state(self):
+        """
+        Return a dictionary of setting toggles.
+        """
+        toggle_states = []
+        for toggle in SettingToggle.get_instances():
+            toggle_state = {
```

Comment:
**Note:** This comment relates to my earlier comment.
1. For consistency, this function could be an `_add_XXX` function.
2. Use `get_or_create_toggle_response` to avoid duplication.
3. Same for setting dicts.